packages/@tailwindcss-upgrade/src/index.ts

```diff
@@ -80,19 +79,44 @@ async function run() {
     // Ensure we are only dealing with CSS files
     files = files.filter((file) => file.endsWith('.css'))
 
-    // Analyze the stylesheets
-    let loadResults = await Promise.allSettled(files.map((filepath) => Stylesheet.load(filepath)))
+    // Load the stylesheets and their imports
+    let sheetsByFile = new Map<string, Stylesheet>()
+    let isIgnored = await isGitIgnored()
+    let queue = files.slice()
+    while (queue.length > 0) {
+      let file = queue.shift()!
 
-    // Load and parse all stylesheets
-    for (let result of loadResults) {
-      if (result.status === 'rejected') {
-        error(`${result.reason}`)
+      // Already handled
+      if (sheetsByFile.has(file)) continue
+
+      // We don't want to process ignored files (like node_modules)
+      if (isIgnored(file)) continue
```

Comment:
When you don't pass any files, then we used globby with the `gitignore: true` option, so I wanted to add that here as well.

I added an explicitly ignored CSS file to the integration tests to make sure we don't touch it.